src/main/java/com/github/cameltooling/lsp/internal/parser/CamelEIPParser.java

```diff
@@ -0,0 +1,51 @@
+package com.github.cameltooling.lsp.internal.parser;
+
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Pattern;
+
+public class CamelEIPParser {
+
+	private final TextDocumentItem document;
+
+	public CamelEIPParser(TextDocumentItem textDocumentItem) {
+		this.document = textDocumentItem;
+	}
+
+	public boolean canPutEIP(Position position) {
+		Pattern CHOICE_EIP_PATTERN = Pattern.compile("\\)\\s*[.[c[h[o[i[c[e]?]?]?]?]?]?]?\\(?$", Pattern.MULTILINE);
```

Comment:
Sonar is complaining of duplicates in the regexp. I guess there should be a simpler way to write it.

Reply:
I'm leaving it out of the iteration for now as RegExr is not complaining nor I can find what would be the alternative right now. Mmm